packages/stream_chat_flutter/lib/src/message_widget.dart

```diff
@@ -97,14 +97,20 @@ class MessageWidget extends StatefulWidget {
     this.deletedBottomRowBuilder,
     this.onReturnAction,
     this.customAttachmentBuilders,
-    this.readList,
     this.padding,
     this.textPadding = const EdgeInsets.symmetric(
       horizontal: 16,
       vertical: 8,
     ),
     this.attachmentPadding = EdgeInsets.zero,
-    this.allRead = false,
+    @Deprecated('''
+    allRead is now deprecated and it will be removed in future releases. 
+    The MessageWidget now listens for read events on its own.
+    ''') this.allRead = false,
+    @Deprecated('''
+    readList is now deprecated and it will be removed in future releases. 
+    The MessageWidget now listens for read events on its own.
+    ''') this.readList,
```

Comment:
The PR looks good. The only issue I see is this: it is breaking without being literal code breaking - we do not use the values of readList after taking it in as a parameter. To make it non breaking, we would likely make it work as it is in default conditions (this is obviously problematic since the current implementation is buggy) or remove the parameters altogether to make it breaking but remove effectively dead parameters.